File: android/app/src/main/java/com/workstationoffice/workcrm/designsystem/L10n.kt
package com.workstationoffice.workcrm.designsystem

import java.util.Locale

/**
 * Bilingual string lookup mirroring the iOS L10n.swift table. Keep additions
 * in sync across both platforms so the QA team only has one translation matrix.
 */
enum class L10n {
    LoginTitle, LoginTenantSlug, LoginEmail, LoginPassword, LoginCta, LoginError,
    TabToday, TabDeals, TabKpi, TabMore,
    VisitsTitle, VisitsEmpty, VisitCheckIn, VisitCheckOut, VisitPendingSync, VisitOffline,
    KpiPersonalTitle, KpiTeamTitle, KpiVisits, KpiWonDeals, KpiRevenue, KpiConversion, KpiDaysLeft,
    CommonRetry, CommonCancel, CommonOk, CommonLoading
}

private val dictionary: Map<L10n, Pair<String, String>> = mapOf(
    L10n.LoginTitle        to ("เข้าสู่ระบบ" to "Sign in"),
    L10n.LoginTenantSlug   to ("รหัสองค์กร" to "Workspace slug"),
    L10n.LoginEmail        to ("อีเมล" to "Email"),
    L10n.LoginPassword     to ("รหัสผ่าน" to "Password"),
    L10n.LoginCta          to ("เข้าสู่ระบบ" to "Sign in"),
    L10n.LoginError        to ("ลงชื่อเข้าใช้ล้มเหลว" to "Sign-in failed"),
    L10n.TabToday          to ("วันนี้" to "Today"),
    L10n.TabDeals          to ("ดีล" to "Deals"),
    L10n.TabKpi            to ("เป้า KPI" to "KPI"),
    L10n.TabMore           to ("เพิ่มเติม" to "More"),
    L10n.VisitsTitle       to ("รายการเยี่ยมวันนี้" to "Today's visits"),
    L10n.VisitsEmpty       to ("ไม่มีนัดในวันนี้" to "No visits scheduled"),
    L10n.VisitCheckIn      to ("เช็คอิน" to "Check in"),
    L10n.VisitCheckOut     to ("เช็คเอาท์" to "Check out"),
    L10n.VisitPendingSync  to ("รอซิงค์" to "Pending sync"),
    L10n.VisitOffline      to ("ออฟไลน์" to "Offline"),
    L10n.KpiPersonalTitle  to ("เป้าหมายของฉัน" to "My KPI"),
    L10n.KpiTeamTitle      to ("เป้าหมายทีม" to "Team KPI"),
    L10n.KpiVisits         to ("จำนวนเยี่ยม" to "Visits"),
    L10n.KpiWonDeals       to ("ดีลที่ปิดได้" to "Won deals"),
    L10n.KpiRevenue        to ("ยอดขาย" to "Revenue"),
    L10n.KpiConversion     to ("อัตราปิดดีล" to "Conversion"),
    L10n.KpiDaysLeft       to ("วันที่เหลือ" to "Days left"),
    L10n.CommonRetry       to ("ลองอีกครั้ง" to "Retry"),
    L10n.CommonCancel      to ("ยกเลิก" to "Cancel"),
    L10n.CommonOk          to ("ตกลง" to "OK"),
    L10n.CommonLoading     to ("กำลังโหลด…" to "Loading…")
)

fun t(key: L10n): String {
    val useThai = Locale.getDefault().language == "th"
    val pair = dictionary[key] ?: return ""
    return if (useThai) pair.first else pair.second
}

File: ios/WorkCRM/DesignSystem/L10n.swift
import Foundation

/// Bilingual string lookup (Thai + English). Workselected.com treats both
/// languages as first-class — we mirror that here. The app auto-picks based on
/// `Locale.current` but the user can override in Settings (future work).
///
/// New strings get added by enum case + a `(th, en)` tuple in `Self.dictionary`.
/// This keeps translations colocated in one file rather than scattered across
/// the codebase's many xliff/Localizable.strings flavors.
public enum L10n: String {
    case loginTitle
    case loginTenantSlug
    case loginEmail
    case loginPassword
    case loginCta
    case loginError

    case tabToday
    case tabDeals
    case tabKpi
    case tabMore

    case dealsScopeAll
    case dealsScopeMine
    case dealsScopeTeam
    case dealsSearchPlaceholder
    case dealsShowClosed
    case dealsOverdueLabel

    case visitsTitle
    case visitsEmpty
    case visitCheckIn
    case visitCheckOut
    case visitPendingSync
    case visitOffline

    case visitFormAddTitle
    case visitFormEditTitle
    case visitFormModePlanned
    case visitFormModeUnplanned
    case visitFormCustomer
    case visitFormCustomerPlaceholder
    case visitFormPlannedAt
    case visitFormObjective
    case visitFormObjectivePlaceholder
    case visitFormUnplannedNote
    case visitFormSave
    case visitFormEdit
    case customerPickerTitle
    case customerPickerSearchPlaceholder
    case customerPickerEmpty
    case customerPickerHintShort

    case kpiPersonalTitle
    case kpiTeamTitle
    case kpiVisits
    case kpiWonDeals
    case kpiRevenue
    case kpiConversion
    case kpiDaysLeft

    case commonRetry
    case commonCancel
    case commonOk
    case commonLoading

    public var localized: String {
        let useThai = Locale.current.language.languageCode?.identifier == "th"
        let pair = Self.dictionary[self] ?? ("", "")
        return useThai ? pair.th : pair.en
    }

    private static let dictionary: [L10n: (th: String, en: String)] = [
        .loginTitle:        ("เข้าสู่ระบบ",            "Sign in"),
        .loginTenantSlug:   ("รหัสองค์กร",             "Workspace slug"),
        .loginEmail:        ("อีเมล",                  "Email"),
        .loginPassword:     ("รหัสผ่าน",                "Password"),
        .loginCta:          ("เข้าสู่ระบบ",            "Sign in"),
        .loginError:        ("ลงชื่อเข้าใช้ล้มเหลว",    "Sign-in failed"),

        .tabToday:          ("วันนี้",                 "Today"),
        .tabDeals:          ("ดีล",                    "Deals"),
        .tabKpi:            ("เป้า KPI",               "KPI"),
        .tabMore:           ("เพิ่มเติม",               "More"),

        .dealsScopeAll:         ("ทั้งหมด",               "All"),
        .dealsScopeMine:        ("ของฉัน",                "Me"),
        .dealsScopeTeam:        ("ทีมของฉัน",             "My team"),
        .dealsSearchPlaceholder:("ค้นหาดีล",              "Search deals"),
        .dealsShowClosed:       ("รวมที่ปิดแล้ว",         "Show closed"),
        .dealsOverdueLabel:     ("ต้องติดตาม",            "Needs follow-up"),

        .visitsTitle:       ("รายการเยี่ยมวันนี้",      "Today's visits"),
        .visitsEmpty:       ("ไม่มีนัดในวันนี้",         "No visits scheduled"),
        .visitCheckIn:      ("เช็คอิน",                "Check in"),
        .visitCheckOut:     ("เช็คเอาท์",              "Check out"),
        .visitPendingSync:  ("รอซิงค์",                 "Pending sync"),
        .visitOffline:      ("ออฟไลน์",                 "Offline"),

        .visitFormAddTitle:             ("สร้างการเยี่ยม",        "New Visit"),
        .visitFormEditTitle:            ("แก้ไขการเยี่ยม",        "Edit Visit"),
        .visitFormModePlanned:          ("วางแผน",                "Planned"),
        .visitFormModeUnplanned:        ("ไม่ได้วางแผน",          "Unplanned"),
        .visitFormCustomer:             ("ลูกค้า",                "Customer"),
        .visitFormCustomerPlaceholder:  ("เลือกลูกค้า",           "Select customer"),
        .visitFormPlannedAt:            ("วัน-เวลา",              "Date & time"),
        .visitFormObjective:            ("วัตถุประสงค์",          "Objective"),
        .visitFormObjectivePlaceholder: ("เช่น นำเสนอผลิตภัณฑ์",  "e.g. Product demo"),
        .visitFormUnplannedNote:        ("จะใช้พิกัดปัจจุบันเพื่อสร้างไซต์ที่ไม่ระบุหากไม่ได้เลือกลูกค้า",
                                         "Current GPS will create an unidentified site if no customer is selected."),
        .visitFormSave:                 ("บันทึก",                "Save"),
        .visitFormEdit:                 ("แก้ไข",                  "Edit"),
        .customerPickerTitle:           ("เลือกลูกค้า",            "Select customer"),
        .customerPickerSearchPlaceholder:("ค้นหาด้วยชื่อหรือรหัส",  "Search by name or code"),
        .customerPickerEmpty:           ("ไม่พบลูกค้า",            "No customers found"),
        .customerPickerHintShort:       ("พิมพ์อย่างน้อย 3 ตัวอักษร", "Type at least 3 characters"),

        .kpiPersonalTitle:  ("เป้าหมายของฉัน",         "My KPI"),
        .kpiTeamTitle:      ("เป้าหมายทีม",            "Team KPI"),
        .kpiVisits:         ("จำนวนเยี่ยม",             "Visits"),
        .kpiWonDeals:       ("ดีลที่ปิดได้",            "Won deals"),
        .kpiRevenue:        ("ยอดขาย",                 "Revenue"),
        .kpiConversion:     ("อัตราปิดดีล",            "Conversion"),
        .kpiDaysLeft:       ("วันที่เหลือ",             "Days left"),

        .commonRetry:       ("ลองอีกครั้ง",             "Retry"),
        .commonCancel:      ("ยกเลิก",                 "Cancel"),
        .commonOk:          ("ตกลง",                   "OK"),
        .commonLoading:     ("กำลังโหลด…",             "Loading…")
    ]
}

/// Quality-of-life: `t(.loginTitle)` reads better than `L10n.loginTitle.localized`.
public func t(_ key: L10n) -> String { key.localized }
